### PR TITLE
[NativeAOT/x86] Implement GetConservativeUpperBoundForOutgoingArgs

### PR DESCRIPTION
**Pull request by filipnavara**
I left this API out when integrating the x86 unwinding code. Turns out it may be necessary for running System.Runtime.Tests.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp [hunk 1]
```diff
@@ -573,9 +573,30 @@ uintptr_t CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(Method
...
+        if (::UnwindStackFrameX86(&registerSet,
+                                  (PTR_CBYTE)(m_moduleBase + pNativeMethodInfo->mainRuntimeFunction->BeginAddress),
+                                  codeOffset,
+                                  &infoBuf,
+                                  table,
+                                  (PTR_CBYTE)(m_moduleBase + pNativeMethodInfo->runtimeFunction->BeginAddress),
+                                  (unwindBlockFlags & UBF_FUNC_KIND_MASK) != UBF_FUNC_KIND_ROOT,
+                                  true))
+        {
+            upperBound = dac_cast<TADDR>(registerSet.PCTAddr - sizeof(TADDR));
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:594]: What is the `- sizeof(TADDR)` for?

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp [hunk 2]
```diff
@@ -573,9 +573,30 @@ uintptr_t CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(Method
...
+        PTR_uint8_t gcInfo;
+        uint32_t codeOffset = GetCodeOffset(pMethodInfo, (PTR_VOID)pRegisterSet->IP, &gcInfo);
+
+        hdrInfo infoBuf;
+        size_t infoSize = DecodeGCHdrInfo(GCInfoToken(gcInfo), codeOffset, &infoBuf);
+        PTR_CBYTE table = gcInfo + infoSize;
+
+        REGDISPLAY registerSet = *pRegisterSet;
+
+        if (::UnwindStackFrameX86(&registerSet,
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:585]: When is this expected to fail?

Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:585]: It’s not, I can change it to ASSERT.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 1 above)
Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:594]: Skipping return address, should have been `+` I guess. I will recheck.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 2 above)
Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp]: Actually, I checked, with funclets enabled it is guaranteed to return `true`.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 1 above)
Comment (filipnavara) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp]: On second look, this (x64) also looks wrong:
https://github.com/dotnet/runtime/blob/825f053e069f1cbb7bc8616693b65e3a98466ec3/src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp#L542-L543

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp]: I do not see a problem with this one.

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**